### PR TITLE
fix: Spark source support table with pattern "db.table"

### LINKED ISSUE #2605: spark source complain about "table or view not found" error.
## Expected Behavior 
spark offline store

## Current Behavior
if set table of SparkSource with pattern "db.table"，feast will complain about "table or view not found" error.

## Steps to reproduce
1. feast init test_repo
2. modify example.py to use SparkSource, and set table of SparkSource with pattern "db.table"
3. configure featue_store.yaml
4. feast apply

### Specifications

- Version: 0.20.1
- Platform: ubuntu 18/04
- Subsystem:

### PR DESCRIPTION
**Pull request by kfiring**
…ps://github.com/feast-dev/feast/issues/2605)

Signed-off-by: leonpeng <kfiring@163.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2605

### CODE REVIEW

**Comment by achals** [comment]
/ok-to-test

Thanks for the PR @kfiring! Could you add a test that verifies this change?

File: sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py [hunk 1]
```diff
@@ -172,7 +175,9 @@ def get_table_query_string(self) -> str:
         """Returns a string that can directly be used to reference this table in SQL"""
         if self.table:
             # Backticks make sure that spark sql knows this a table reference.
-            return f"`{self.table}`"
+            # return f"`{self.table}`"
+            table = '.'.join(map(lambda x: f'`{x}`', self.table.split('.')))
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py:179]: nit, `map` and `lambda` are generally avoided in favor of list comprehensions:
``
```suggestion
            table = '.'.join([f"`{x}`" for x in self.table.split('.')])
```

**Comment by achals** [comment]
Thanks for the PR @kfiring! Do you mind addressing lint/DCO?

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**